### ISSUE TITLE
Compilation error with .NET 4.0

### ISSUE THREAD

**Issue by UweKeim** [AngleSharp/AngleSharp#307]
When using AngleSharp (via NuGet) in a .NET 4.0 project, I get the error:

> The primary reference "AngleSharp, Version=0.9.5.41764, Culture=neutral, PublicKeyToken=e83494dcdc6d31ea, processorArchitecture=MSIL" could not be resolved because it has an indirect dependency on the framework assembly "System.Threading.Tasks, Version=2.6.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" which could not be resolved in the currently targeted framework. ".NETFramework,Version=v4.0". To resolve this problem, either remove the reference "AngleSharp, Version=0.9.5.41764, Culture=neutral, PublicKeyToken=e83494dcdc6d31ea, processorArchitecture=MSIL" or retarget your application to a framework version which contains "System.Threading.Tasks, Version=2.6.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".

My understanding is that the .NET 4.0 package of AngleSharp actually has implicit references to functionality that is available only in higher versions of the .NET Framework.

**My question:**

Any chance to get AngleSharp working with pure .NET 4.0 only?

**Comment by FlorianRappl** [comment]
Seems like a problem with NuGet. The NuGet package contains a version specifically targeting .NET 4 (here the Bcl.Async package is used to resolve the dependencies for the async-state-machinery).

**Comment by UweKeim** [comment]
Thanks.

Any chance for me to manually resolve this?

Frohe Ostern aus Stuttgart :smile:

**Comment by FlorianRappl** [comment]
Could you have a look what path has been resolved (just to confirm the problem)? Instead of pointing to _.../net40/AngleSharp.dll_ I suspect it points to _.../net45/AngleSharp.dll_ or _../portable-windows8+net45+windowsphone8+wpa/AngleSharp.dll_.

The simplest thing would be to change that path manually, however, I don't know if NuGet will realize and revoke that change. If so you could remove it from the _packages.config_ for the project.

However, in general I would like to improve the situation (the targets are pretty clear, so I am not sure if the problem is really on my side, or if its just a flaw in NuGet). So if anyone has a clue why NuGet fails to resolve the right target I would welcome any PRs or suggestions. This may be related to #85.

Frohe Ostern aus München :rabbit:

**Comment by UweKeim** [comment]
Now this is strange:

I've uninstalled everything to the initial state (i.e. before adding AngleSharp and the BCL packages).

Then I've restarted VS.NET 2015 and added the three BCL libraries first, then a suggested update to the BCL library. 

After this update, it says:

>  Failed to uninstall 'Microsoft.Bcl.Build.1.0.14'. Restart Visual Studio to finish this process.

I did as suggested, after a restart I installed AngleSharp.

Now there are _no_ more compilation errors.

Probably my error was that I initially ignored the "...Restart Visual Studio..." suggestion.

**tl;dr** Everything works fine now on .NET 4.0

Thanks for your fast public holiday help and sorry for reporting something that was my own fault!

**Comment by FlorianRappl** [comment]
Alright - no problem. Thanks for investigating!

**Comment by UweKeim** [comment]
As a side-note, this seems to be related to [this issue](https://github.com/StackExchange/StackExchange.Redis/issues/96).
